### PR TITLE
Add detailed output about dependency conflicts

### PR DESCRIPTION
**Pull request by arjantop**
In case of depenency conflicts output more detailed information about
which depenency requested which version.
This makes resolving dependency conflicts easier.

### CODE REVIEW

**Comment by arjantop** [comment]
Fixes #106

File: src/scala/com/github/johnynek/bazel_deps/Writer.scala [hunk 1]
```diff
@@ -82,10 +82,14 @@ object Writer {
           case Some(vs) =>
             val status =
               if (isRoot) s"fixed to ${v.asString}"
-              else if (vs.max == v) s"promoted to ${v.asString}"
+              else if (vs.map(_.destination.version).max == v) s"promoted to ${v.asString}"
               else s"downgraded to ${v.asString}"
 
-            s"""# duplicates in ${coord.unversioned.asString} $status. Versions: ${vs.toList.sorted.map(_.asString).mkString(" ")}\n"""
+            val sb = new StringBuilder(s"""# duplicates in ${coord.unversioned.asString} $status\n""")
```
Comment (johnynek) [inline src/scala/com/github/johnynek/bazel_deps/Writer.scala]: Minor style request. I appreciate not doing naive String concatenation but actually scala’s mkString(prefix, sep, end) method on Iterable also uses StringBuilder.

Can you change foreach to map followed by mkString and avoid an explicit mutable builder?

Comment (arjantop) [reply src/scala/com/github/johnynek/bazel_deps/Writer.scala]: Done, please check the "fix" when you have time.

I force pushed, I hope it is ok.

**Comment by johnynek** [comment]
Merged! Thank you.

**Pull request merged**